tools/src/test/java/org/apache/kafka/tools/FeatureCommandTest.java

```diff
@@ -288,13 +335,42 @@ public void testHandleDisableDryRun() {
             "Can not disable quux. Invalid update version 0 for feature metadata.version. Can't downgrade below 4"), disableOutput);
     }
 
+    @Test
+    public void testInvalidReleaseVersion() {
+        Map<String, Object> namespace = new HashMap<>();
+        namespace.put("release_version", "foo");
+        ToolsTestUtils.captureStandardOut(() -> {
+            Throwable t = assertThrows(TerseException.class, () -> FeatureCommand.handleUpgrade(new Namespace(namespace), buildAdminClient()));
+            assertTrue(t.getMessage().contains("Unknown metadata.version foo."));
+        });
+    }
+
+    @Test
+    public void testIncompatibleUpgradeFlags() {
+        Map<String, Object> namespace = new HashMap<>();
+        namespace.put("release_version", "3.3-IV3");
+        namespace.put("feature", Arrays.asList("foo.bar", "metadata.version", "quux"));
+        ToolsTestUtils.captureStandardOut(() -> {
+            Throwable t = assertThrows(TerseException.class, () -> FeatureCommand.handleUpgrade(new Namespace(namespace), buildAdminClient()));
+            assertTrue(t.getMessage().contains("Can not specify `release-version` with other feature flags."));
+        });
+
+
+        namespace.put("release_version", "3.3-IV3");
+        namespace.put("metadata", "3.3-IV3");
+        ToolsTestUtils.captureStandardOut(() -> {
+            Throwable t = assertThrows(TerseException.class, () -> FeatureCommand.handleUpgrade(new Namespace(namespace), buildAdminClient()));
+            assertTrue(t.getMessage().contains("Can not specify `release-version` with other feature flags."));
+        });
+    }
+
     @Test
     public void testHandleVersionMappingWithValidReleaseVersion() {
         Map<String, Object> namespace = new HashMap<>();
         namespace.put("release_version", "3.3-IV3");
         String versionMappingOutput = ToolsTestUtils.captureStandardOut(() -> {
             try {
-                FeatureCommand.handleVersionMapping(new Namespace(namespace));
+                FeatureCommand.handleVersionMapping(new Namespace(namespace), testingFeatures);
```

Comment:
Why do we need to include the testing feature when it's never used in the tool?

Reply:
We don't need to test it, but I thought it could be useful to test alongside the dependencies check. I can remove it if it seems like too much extra complexity.